WIP: Private key management chapter (#48)
BitcoinDesign/Guide: guide/private-key-management/case-studies.md

- [Current / Checking account]({{ '/guide/private-key-management/case-studies/#current--checking-account' | relative_url }})
- [General upgradeable wallet]({{ '/guide/private-key-management/case-studies/#general-upgradeable-wallet' | relative_url }})
- [Shared account]({{ '/guide/private-key-management/case-studies/#shared-account' | relative_url }})

***

## Cash / Daily spending

Imagine a product which tries to solve the problem of quickly and easily sending smaller amounts of money to friends and family, or for small purchases. Ease and speed of use will be important as usage is likely to be on mobile devices and on the go. Users are not expected to be well versed in bitcoin technology or advanced private key management, which makes it reasonable to worry more about self-inflicted loss than from theft.

A single-key scheme with [automatic cloud backup]({{ '/guide/private-key-management/single-user-schemes/#automatic-cloud-backup' | relative_url }}) might be the best choice for the majority of users in this case. For more advanced users you could offer the option to opt-out of automatic cloud backup and instead use a recovery-phrase.

> <iframe style="border: 0px solid rgba(0, 0, 0, 0.1);" width="450" height="1000" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Fproto%2FSRWlaxbDulsacpPQn2TTri%2FCase-study-prototypes%3Fnode-id%3D1%253A3%26viewport%3D1204%252C392%252C0.5%26scaling%3Dmin-zoom" allowfullscreen></iframe>

Author: Good point. 
Not sure what to here as it's mostly in Figma's hands.
Options:

1. Accept load times, not helped by having several on one page
2. Accept load times, but split case studies into separate pages
3. Don't embed, show static image or gif and then link to prototype externally

> <iframe style="border: 0px solid rgba(0, 0, 0, 0.1);" width="450" height="1000" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Fproto%2FSRWlaxbDulsacpPQn2TTri%2FCase-study-prototypes%3Fnode-id%3D1%253A3%26viewport%3D1204%252C392%252C0.5%26scaling%3Dmin-zoom" allowfullscreen></iframe>

Reviewer: As far as an image goes, how about we include all screens in the prototype laid out next to each other with arrows indicating the sequence? That allows readers to easily understand the full user flow, and we can still offer a "View prototype" option to feel out the interaction.

> <iframe style="border: 0px solid rgba(0, 0, 0, 0.1);" width="450" height="1000" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Fproto%2FSRWlaxbDulsacpPQn2TTri%2FCase-study-prototypes%3Fnode-id%3D1%253A3%26viewport%3D1204%252C392%252C0.5%26scaling%3Dmin-zoom" allowfullscreen></iframe>

Reviewer: Also think the prototype needs some explanation. "The following prototype shows how convenient a cloud backup user flow can be. Start by tapping 'Create wallet'." Prototypes are great to make designs feel more real, but they don't explain themselves at all.

> <iframe style="border: 0px solid rgba(0, 0, 0, 0.1);" width="450" height="1000" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Fproto%2FSRWlaxbDulsacpPQn2TTri%2FCase-study-prototypes%3Fnode-id%3D1%253A3%26viewport%3D1204%252C392%252C0.5%26scaling%3Dmin-zoom" allowfullscreen></iframe>

Reviewer: "Recieve" should be "Receive" in the prototype.

A single-key scheme with [automatic cloud backup]({{ '/guide/private-key-management/single-user-schemes/#automatic-cloud-backup' | relative_url }}) might be the best choice for the majority of users in this case. For more advanced users you could offer the option to opt-out of automatic cloud backup and instead use a recovery-phrase.

<iframe style="border: 0px solid rgba(0, 0, 0, 0.1);" width="450" height="1000" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Fproto%2FSRWlaxbDulsacpPQn2TTri%2FCase-study-prototypes%3Fnode-id%3D1%253A3%26viewport%3D1204%252C392%252C0.5%26scaling%3Dmin-zoom" allowfullscreen></iframe>

#### Design considerations 
- Most users will be bitcoin beginners
- Quick and easy send/receive
- Onboarding with as little friction as possible

#### Technical considerations
- Back up encrypted recovery-phrase automatically to retain export option in the future
- Use a platform-appropriate storage location (keychain with iCloud, Google Drive)

> - Additional user authentication to consider: biometrics, pin)

Reviewer: ```suggestion
- Additional user authentication to consider: biometrics, pin
```

***

## Current / Checking account

A product that is meant to be a replacement for what a bank would call a current/checking account where the user might receive their salary and pay their bills from. 
Safeguards against loss will be a higher priority than with a cash product, and we might therefore accept more friction both when setting up the wallet and when transacting.
If users have no prior bitcoin knowledge we should expect to spend a significant effort educating them to put them in a position to safely operate the wallet product.

A 2-of-3 [multi-key setup]({{ '/guide/private-key-management/single-user-schemes/#multi-key' | relative_url }}) would seem the most appropriate here, although it will be a significant hurdle in onboarding. Other schemes could be considered but come with distinct downsides for amounts of value we can expect users to store in this use-case. A single-key scheme with an automatic cloud backup, recovery-phrase or single signing device could work at the lower end of the value scale, but start to look like less responsible recommendations with higher values due to their single points of failure.

> A key question is the combination of key-storage devices and their distribution. We have many options here, and it might come down to the experience of the target audience and their expected access to the necessary hardware; 

Reviewer: ```suggestion
An important question is the combination of key-storage devices and their distribution. We have many options here, and it might come down to the experience of the target audience and their expected access to the necessary hardware; 
```

The idea here is that the user will control the wallet through an app on their main mobile device, but when they are sending funds (paying bills etc) they will need to confirm the transaction on the purpose-built signing device.

Should they lose either their main mobile device, or the purpose-built signing device they can replace the lost key (rotate in a new key) with the help of the recovery key. However, if they lose both the mobile and signing device they will not be able to recover their funds, unless they had also backed up either of the respective recovery-phrases. 

<iframe style="border: 0px solid rgba(0, 0, 0, 0.1);" width="450" height="1000" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Fproto%2FSRWlaxbDulsacpPQn2TTri%2FCase-study-prototypes%3Fnode-id%3D6%253A11%26viewport%3D479%252C82%252C0.5%26scaling%3Dscale-down" allowfullscreen></iframe>

#### Design considerations
- Suitable for monthly transactions
- Lots of edge cases and infrequent but important situations
- Key setup, rotation, recovery and signing flows

#### Technical considerations

> - Pitfalls of using hardware wallets in multisig

Reviewer: Do we have info on what those pitfalls are?

Author: It was a reference (although not linked) to this article: https://medium.com/shiftcrypto/the-pitfalls-of-multisig-when-using-hardware-wallets-9b0e98e4c19c

But the real solution here is to write a comprehensive guide to what is required to backup multi-sigs, I haven't yet found one. I'll rephrase to unblock for now.

> - Need to take care when backing up to include x-pubs from all keys

Reviewer: I'd suggest writing out "Extended Public Keys".

What is meant by needing to take care... Is that a developer task?

Author: Rephrased, like above, solution is to write proper Multisig backup instructions

> - Usage could be simplified bye having hardware device designed/customized by the software maker (compare banks and their digital token signers common in Europe).

Reviewer: ```suggestion
- Usage could be simplified by having hardware device designed/customized by the software maker (compare banks and their digital token signers common in Europe)
```

You can find a prototype for this case study embedded above, and a Figma file with [outline designs here](https://www.figma.com/file/SRWlaxbDulsacpPQn2TTri/Case-study-prototypes?node-id=6%3A9){:target="_blank"}.

***

## General upgradeable wallet

Although it is generally easier to build a great experience with a specific use-case in mind, let's look at a case where we would like to make a wallet that is as general as possible. It needs to be suitable both for beginners and expanding users, and for holding anything between small and significant amounts. How do we choose a single private key management scheme for this situation?

> The solution in this case could be a wallet that enables the user to upgrade the scheme as their experience, and funds grow. The idea is to provide progressive security that doesn't introduce unnecessary friction until it is required. Our wallet will be able to switch from;

Reviewer: ```suggestion
The solution in this case could be a wallet that enables the user to upgrade the scheme as their experience and funds grow. The idea is to provide progressive security that doesn't introduce unnecessary friction until it is required. Our wallet will be able to switch from;
```
